src/components/data/DataNavigation.js

```diff
@@ -367,44 +396,19 @@ function DataNavigation(props) {
     }, [dataRoots, handlePathChange, path]);
 
     const { error } = useQuery({
-        queryKey: ["dataFileSystemRoots", userProfile],
+        queryKey: "dataFileSystemRoots",
         queryFn: getFilesystemRoots,
         config: {
-            onSuccess: (respData) => {
-                if (respData && userProfile) {
-                    const respRoots = respData.roots;
-                    const home = respRoots.find(
-                        (root) => root.label === userProfile.id
-                    );
-                    home.icon = <HomeIcon />;
-                    const sharedWithMe = respRoots.find(
-                        (root) => root.label === constants.SHARED_WITH_ME
-                    );
-                    setSharedWithMePath(sharedWithMe.path);
-                    sharedWithMe.icon = <FolderSharedIcon />;
-                    const communityData = respRoots.find(
-                        (root) => root.label === constants.COMMUNITY_DATA
-                    );
-                    setCommunityDataPath(communityData.path);
-                    communityData.icon = <GroupIcon />;
-                    const trash = respRoots.find(
-                        (root) => root.label === constants.TRASH
-                    );
-                    trash.icon = <DeleteIcon />;
-
-                    const basePaths = respData["base-paths"];
-                    setUserHomePath(basePaths["user_home_path"]);
-                    setUserTrashPath(basePaths["user_trash_path"]);
-                    setDataRoots([home, sharedWithMe, communityData, trash]);
-                }
-            },
+            onSuccess: preProcessData,
             onError: () => {
                 //temporary workaround -> should still display community data
                 announce({
                     text: formatMessage(intl, "dataSignIn"),
                     variant: AnnouncerConstants.INFO,
                 });
             },
+            staleTime: Infinity,
+            cacheTime: Infinity,
```

Comment:
This setting will make `react-query` cache the roots. If you open, the react-query  dev tools. You can see this in action.